.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,86 @@
+## Type of change
+
+<!-- Mandatory
+Please delete options that are not relevant
+-->
+
+- [ ] New feature
+- [ ] Bug fix
+- [ ] Enhancement
+- [ ] Breaking change
+
+## What does this PR do?
+
+<!-- Mandatory
+Explain here the changes you made on the PR. Please explain the WHAT: patterns used, algorithms implemented, design architecture, message processing, etc.
+-->
+
+## Why is it important?
+
+<!-- Mandatory
+Explain here the WHY, or the rationale/motivation for the changes.
+-->
+
+## Checklist
+
+<!-- Mandatory
+Add a checklist of things that are required to be reviewed in order to have the PR approved
+
+List here all the items you have verified BEFORE sending this PR. Please DO NOT remove any item, striking through those that do not apply.
+-->
+
+- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/). 
+- [ ] My code follows the style guidelines of this project
+- [ ] I have performed a self-review of my own code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing **unit** tests pass locally with my changes
+- [ ] New and existing **integration** tests pass locally with my changes
+- [ ] Any dependent changes have been merged and published in downstream modules
+- [ ] New dependencies have been vendorised
```

Comment:
I'd like the team to discuss about the items in the check list

Reply:
We should add `I have made corresponding changes in the default and reference configuration files`.

How about thinning out this list as well. Each PR should be run via CI. Only if CI is green shall the PR be merged. The intake job calls `make check` to validate style (go fmt, autopep8, go vet). Running integration tests requires some time. Although it is appreciated I would not expect devs to run the complete test suite (up to integration tests) upfront.

Items I personally would remove from the list:
- New and existing **unit** tests pass locally with my changes  : the moment we open the PR CI will run tests anyways. Everyone will be able to see if unit tests pass.
- New and existing **integration** tests pass locally with my changes : Same as unit tests
- New dependencies have been vendorised : The build/CI will fail anyways if that is not the case.

Items I might consider to remove from the list:
- I have signed the : If the CLA checker fails it should print a message + link anyways.
- My changes generate no new warnings. : The go compiler doesn't really generate warnings (it's more oppioninated). And even for C/C++ I'm in favor of using -werror :) . Any other warnings we have in mind?


What exactly is meant by `Any dependent changes have been merged and published in downstream modules`? Would CI be able to pass if this is not the case?